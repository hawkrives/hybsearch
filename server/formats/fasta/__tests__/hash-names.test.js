/* eslint-env jest */
'use strict'

const expect = require('expect')
const hashNames = require('../hash-names')

test('shortens the accession identifiers', () => {
	let input = `> Emydura_subglobosa__KC755190
ggaacaataaattatcacctcaaaagacac
> Emydura_victoriae__KC755189
ggaacaatcaattattaccccacaaagac
> Emydura_victoriae__KC755188
ggaacaatcaattattaccccacaaggaa`

	let hashedNames = hashNames(input)

	expect(hashedNames).toMatchSnapshot()
})

test('does not pad accession numbers to the left with 0s when they grow past one digit', () => {
	let input = `> Emydura_victoriae__KC755189
ggaacaataaattatcacctcaaaagacac
> Emydura_victoriae__KC755189
ggaacaatcaattattaccccacaaagac
> Emydura_victoriae__KC755188
ggaacaatcaattattaccccacaaggaa
> Emydura_victoriae__KC755188
ggaacaatcaattattaccccacaaggaa
> Emydura_victoriae__KC755188
ggaacaatcaattattaccccacaaggaa
> Emydura_victoriae__KC755188
ggaacaatcaattattaccccacaaggaa
> Emydura_victoriae__KC755188
ggaacaatcaattattaccccacaaggaa
> Emydura_victoriae__KC755188
ggaacaatcaattattaccccacaaggaa
> Emydura_victoriae__KC755188
ggaacaatcaattattaccccacaaggaa
> Emydura_victoriae__KC755188
ggaacaatcaattattaccccacaaggaa
> Emydura_victoriae__KC755188
ggaacaatcaattattaccccacaaggaa
> Emydura_victoriae__KC755188
ggaacaatcaattattaccccacaaggaa
> Emydura_victoriae__KC755188
ggaacaatcaattattaccccacaaggaa
> Emydura_victoriae__KC755188
ggaacaatcaattattaccccacaaggaa
> Emydura_victoriae__KC755188
ggaacaatcaattattaccccacaaggaa`

	let hashedNames = hashNames(input)

	for (let key of Object.keys(hashNames)) {
		expect(key).toMatch(/ahjmjgex[1-9][0-9]*/)
	}

	expect(hashedNames).toMatchSnapshot()
})
