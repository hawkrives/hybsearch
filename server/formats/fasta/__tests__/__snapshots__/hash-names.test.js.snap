// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pads accession numbers to the left with 0s when they grow past one digit 1`] = `
Object {
  "ahjmjgex01": "Emydura_victoriae__KC755189",
  "ahjmjgex02": "Emydura_victoriae__KC755189",
  "ahjmjgex03": "Emydura_victoriae__KC755188",
  "ahjmjgex04": "Emydura_victoriae__KC755188",
  "ahjmjgex05": "Emydura_victoriae__KC755188",
  "ahjmjgex06": "Emydura_victoriae__KC755188",
  "ahjmjgex07": "Emydura_victoriae__KC755188",
  "ahjmjgex08": "Emydura_victoriae__KC755188",
  "ahjmjgex09": "Emydura_victoriae__KC755188",
  "ahjmjgex10": "Emydura_victoriae__KC755188",
  "ahjmjgex11": "Emydura_victoriae__KC755188",
  "ahjmjgex12": "Emydura_victoriae__KC755188",
  "ahjmjgex13": "Emydura_victoriae__KC755188",
  "ahjmjgex14": "Emydura_victoriae__KC755188",
  "ahjmjgex15": "Emydura_victoriae__KC755188",
}
`;

exports[`shortens the accession identifiers 1`] = `
Object {
  "ahjmjgejx1": "Emydura_victoriae__KC755189",
  "ahjmjgejx2": "Emydura_victoriae__KC755188",
  "blgcgdgcx1": "Emydura_subglobosa__KC755190",
}
`;
